server/src/main/java/org/elasticsearch/index/query/DistanceFeatureQueryBuilder.java

```diff
@@ -112,7 +112,8 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
         if (fieldType == null) {
             return Queries.newMatchNoDocsQuery("Can't run [" + NAME + "] query on unmapped fields!");
         }
-        return fieldType.distanceFeatureQuery(origin.origin(), pivot, boost, context);
+        // As we already apply boost in AbstractQueryBuilder::toQuery, we always passing a boost of 1.0 to distanceFeatureQuery
+        return fieldType.distanceFeatureQuery(origin.origin(), pivot, 1.0f, context);
```

Comment:
When we fix `LongScriptFieldDistanceFeatureQuery`, I think we could remove the `boost` parameter altogether from `MappedFieldType#distanceFeatureQuery` ?

Reply:
Good point, I or Nik will do that.